x-pack/plugins/maps/public/components/single_field_select.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import React from 'react';
+
+import {
+  EuiComboBox,
+  EuiComboBoxProps,
+  EuiComboBoxOptionOption,
+  EuiHighlight,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiToolTip,
+} from '@elastic/eui';
+import { IFieldType } from 'src/plugins/data/public';
+import { FieldIcon } from '../../../../../src/plugins/kibana_react/public';
+
+function fieldsToOptions(
+  fields?: IFieldType[],
+  isFieldDisabled?: (field: IFieldType) => boolean
```

Comment:
Consider not using optional params at all.  This isn't a utility for 3rd parties, so I'd do the justification closer to where this is being used.

https://github.com/nreese/kibana/blob/f125deacaad3b69aed6263d50233c2ed01fe5fbc/x-pack/plugins/maps/public/components/single_field_select.tsx#L110

```
      options={fieldsToOptions(fields || [], isFieldDisabled || () => false)}
```